sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -363,14 +363,28 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
     /**
      * @brief HTTP retry policy.
      */
-    class RetryPolicy
-#if !defined(_azure_TESTING_BUILD)
-        final
-#endif
-        : public HttpPolicy {
+    class RetryPolicy : public HttpPolicy {
     private:
       RetryOptions m_retryOptions;
 
+#if defined(_azure_TESTING_BUILD)
+    protected:
+#else
+    private:
+#endif
+      virtual bool ShouldRetryOnTransportFailure(
```

Comment:
These two functions are virtual just for testing, right?

Reply:
Yes. I actually want to make them non-virtual/private and update the tests, at some point.